### PR TITLE
8255381: com/sun/jdi/EATests.java should not suspend graal threads

### PR DESCRIPTION
**Pull request by reinrich**
This fixes a bug in the test test/jdk/com/sun/jdi/EATests.java that caused
timeout failures when graal is enabled.

The fix is to avoid suspending all threads when a breakpoint is reached and then resume
just the main thread again. This pattern was used in the test case
EAMaterializeLocalAtObjectPollReturnReturn. It caused timeouts because graal
threads remained suspended and, running with -Xbatch, the main thread waited
(with timeout) for completion of compile tasks.
The fix was applied to all breakpoints in the test. All explicit suspend calls now apply only
to the main test thread and all explicit resume calls apply to all java threads.

Testing: duration of the test case EAMaterializeLocalAtObjectPollReturnReturn is
reduced from 30s to 10s.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8255381](https://bugs.openjdk.java.net/browse/JDK-8255381): com/sun/jdi/EATests.java should not suspend graal threads

### Reviewers
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1625/head:pull/1625`
`$ git checkout pull/1625`

### CODE REVIEW

File: test/jdk/com/sun/jdi/EATests.java [hunk 1]
```diff
@@ -1248,17 +1261,21 @@ public int getExpectedIResult() {
...
+        int retryCount = 0;
         do {
             env.targetMainThread.suspend();
             printStack(env.targetMainThread);
             try {
                 o = getLocalRef(env.targetMainThread.frame(0), XYVAL_NAME, "xy");
             } catch (Exception e) {
-                msg("The local variable xy is out of scope because we suspended at the wrong bci. Resume and try again!");
-                env.targetMainThread.resume();
+                msg("The local variable xy is out of scope because we suspended at the wrong bci. Resume and try again! (" + (++retryCount) + ")");
```
[the remark this answers is not in the document]
Comment (reinrich) [reply test/jdk/com/sun/jdi/EATests.java]: Sure.

File: test/jdk/com/sun/jdi/EATests.java [hunk 2]
```diff
@@ -1248,17 +1261,21 @@ public int getExpectedIResult() {
...
+                env.vm().resume();
```
[the remark this answers is not in the document]
Comment (reinrich) [reply test/jdk/com/sun/jdi/EATests.java]: Ok, that's probably better.
(I wanted to follow the principle to always resume _all_ threads)

**Review by plummercj — APPROVED** [review]

**Comment by reinrich** [comment]
Thanks for the review @plummercj

**Review by sspitsyn — APPROVED** [review]
Hi Richard,
The fix looks okay to me.
Thanks,
Serguei

**Comment by reinrich** [comment]
Hi Serguei,
thanks a lot for reviewing this.
Richard.

**Comment by reinrich** [comment]
@plummercj, @sspitsyn, thanks again for the review.

I would like to bring the fix to jdk16. According to https://openjdk.java.net/jeps/3 this is possible in RDP1 for test bug fixes without approval.

I'll try to open a PR against jdk16. Would be great if you could review that one too.

Thanks, Richard.

**Comment by reinrich** [comment]
I'm closing this PR since the change was integrated in jdk16 with openjdk/jdk16#7

**Pull request closed without merging**